packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -124,6 +127,24 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       return undefined;
     }
 
+    tsParseModifiers<T: TsModifier>(allowedModifiers: T[]): { [*]: true } {
+      const modifiers = {};
```

Comment:
I'd feel better with `Object.create(null)` but it probably isn't really relevant here.